File: tests/phase4/lib_runtime_fixtures.py
# -*- coding: utf-8 -*-
"""أدوات مشتركة لاختبارات المرحلة 4 في بايثون — نظير lib_runtime_fixtures.js."""
import json
import os

HERE = os.path.dirname(os.path.abspath(__file__))
ROOT = os.path.dirname(os.path.dirname(HERE))


def load():
    with open(os.path.join(HERE, 'fixtures', 'runtime_scenarios.json'), encoding='utf-8') as f:
        return json.load(f)


def hydrate(v):
    if isinstance(v, list):
        return [hydrate(x) for x in v]
    if isinstance(v, dict):
        return {k: hydrate(x) for k, x in v.items()}
    if v == 'NaN_MARKER':
        return float('nan')
    if v == 'INF_MARKER':
        return float('inf')
    if v == 'NEG_INF_MARKER':
        return float('-inf')
    return v

File: tests/phase4/lib_runtime_fixtures.js
/* أدوات مشتركة لاختبارات المرحلة 4: تحميل التجهيزات واستبدال علامات القيم
   غير المنتهية بقيم حقيقية (JSON لا يستطيع حمل NaN أو Infinity). */
const fs=require('fs'), _np=require('path');
const HERE=__dirname, ROOT=_np.resolve(HERE,'..','..');
function load(){
  return JSON.parse(fs.readFileSync(_np.join(HERE,'fixtures','runtime_scenarios.json'),'utf8')); }
function hydrate(v){
  if(Array.isArray(v)) return v.map(hydrate);
  if(v&&typeof v==='object'){ const o={};
    Object.keys(v).forEach(k=>{o[k]=hydrate(v[k]);}); return o; }
  if(v==='NaN_MARKER') return NaN;
  if(v==='INF_MARKER') return Infinity;
  if(v==='NEG_INF_MARKER') return -Infinity;
  return v; }
module.exports={load,hydrate,ROOT,HERE};
